persistent/Database/Persist/Sql/Class.hs

```diff
@@ -58,6 +58,13 @@ instance PersistField a => RawSql (Single a) where
     rawSqlProcessRow [pv]  = Single <$> fromPersistValue pv
     rawSqlProcessRow _     = Left $ pack "RawSql (Single a): wrong number of columns."
 
+instance (PersistEntity a, PersistEntityBackend a ~ SqlBackend) => RawSql (Key a) where
+  rawSqlCols _ key         = (length $ keyToValues key, [])
```

Comment:
Is it inefficient to use `keyToValues` just to determine the number of columns? Didn't come up with a way to get the column count otherwise.